src/components/Attachment/AttachmentContainer.tsx

```diff
@@ -72,11 +81,41 @@ export const GalleryContainer = <
 >({
   attachment,
   Gallery = DefaultGallery,
-}: RenderGalleryProps<StreamChatGenerics>) => (
-  <AttachmentWithinContainer attachment={attachment} componentType='gallery'>
-    <Gallery images={attachment.images || []} key='gallery' />
-  </AttachmentWithinContainer>
-);
+}: RenderGalleryProps<StreamChatGenerics>) => {
+  const imageElements = useRef<HTMLElement[]>([]);
+  const { imageAttachmentSizeHandler } = useChannelStateContext();
+  const [attachmentConfigurations, setAttachmentConfigurations] = useState<
+    ImageAttachmentConfigration[]
+  >([]);
+
+  useEffect(() => {
```

Comment:
I am thinking that for adjusting the element styles should be used `useLayoutEffect` but not sure, whether that caused you some problems.

Reply:
I was thinking the same, maybe it will be better for the virtualized mode.